### PR TITLE
require peaks

### LINKED ISSUE #90: add equivalent filter of min_peaks
https://github.com/matchms/matchms/blob/61602e744be587be98122d5964d5ed75cca2ecba/matchms/old/ms_functions.py#L69-L71

### PR DESCRIPTION
**Pull request by jspaaks**
refs #90 

This issue adds a new filter, ``require_minimum_number_of_peaks``. It is similar but not equal to https://github.com/matchms/old-iomega-spec2vec/blob/216b8f8b5e4ffd320b4575326a05fb6c7cd28223/matchms/old/ms_functions.py#L591-L593

Here are some features of this implementation:
1. The absolute minimum required peaks can be set by a user via the ``n_required=10`` parameter; this condition needs to met at all times.
1. The minimum required peaks as related to mz of parent can be set separately via the ``ratio_required`` parameter (in the old implementation it was related to the absolute threshold). If set, the condition needs to be met at all times
1. The filter returns ``None`` for non-qualifying spectrums 

The implication of this last item is that I adapted all filters to be able to deal with input spectrums being None, and that the user code should include a list comprehension along the lines of 

```python
spectrums = [s for s in spectrums if s is not None]
```

### CODE REVIEW

File: matchms/filtering/require_minimum_number_of_peaks.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from math import ceil
+
+
+def require_minimum_number_of_peaks(spectrum_in, n_required=10, ratio_required=None):
+
```
Comment (florian-huber) [inline matchms/filtering/require_minimum_number_of_peaks.py:5]: I think that the parent_mass based scaling is not self-explaining. We could of course also start adding the docstrings in a separate PR.

**Review by florian-huber — APPROVED** [review]
Looks nice, thanks Juriaan!
I only left one comment regarding docstrings. It feels to me like we should add more of them soonish to not loose sight.

**Comment by jspaaks** [comment]
> Looks nice, thanks Juriaan!
> I only left one comment regarding docstrings. It feels to me like we should add more of them soonish to not loose sight.

Agree, but that means addressing https://github.com/matchms/matchms/issues/128

File: matchms/filtering/require_minimum_number_of_peaks.py (hunk 1 above)
Comment (jspaaks) [reply matchms/filtering/require_minimum_number_of_peaks.py:5]: Agree, it is not immediately clear why that option is there. I propose that we do a round of only working on docstrings later, preferably after we address #128.

File: matchms/Scores.py [hunk 2]
```diff
@@ -60,15 +64,20 @@ def __init__(self, references: ReferencesType, queries: QueriesType, similarity_
             List of query objects
         similarity_function
             Function which accepts a reference + query object and returns a score or tuple of scores
+        is_symmetric
+            Set to True when references and queries are identical (as for instance for an all-vs-all
+            comparison). By using the fact that score[i,j] = score[j,i] the calculation will be about
```
Comment (sverhoeven) [inline matchms/Scores.py:70]: Also the similarity_function must give same result when you call with either (a,b) or (b,a).

**Pull request merged**